src/op_analytics/coreutils/time.py

```diff
@@ -64,3 +64,10 @@ def surrounding_dates(dateval: date) -> list[date]:
 def epoch_is_date(epoch: int) -> bool:
     """Return true if seconds since epoch is at exactly midnight."""
     return epoch % 86400 == 0
+
+
+def parse_isoformat_with_z(iso_string):
```

Comment:
nice, this is great to have! 

I noticed that the Z suffix does work out of the box:
```
In [1]: from datetime import datetime

In [2]: datetime.fromisoformat("2023-07-09T13:47:17.000Z")
Out[2]: datetime.datetime(2023, 7, 9, 13, 47, 17, tzinfo=datetime.timezone.utc)
```
but maybe I'm missing something about an edge case? If so we should comment why we handle the Z as a special case. 

Also it's important to note that this method is returning a datetime with timezone. Using fromisoformat can be tricky since in some cases it returns tzinfo and in some cases it does not. 
```
In [5]: datetime.fromisoformat("2023-07-09T13:47:17.000").tzinfo

In [6]: datetime.fromisoformat("2023-07-09T13:47:17.000Z").tzinfo
Out[6]: datetime.timezone.utc
```

Functions in the time module always return naive datetimes (all callers can safely assume the object is UTC and should not ever have to worry about handling timezones), so wondering if here we would prefer to ensure that we return a naive datetime object always.

Reply:
hey! I see the function now returns a datetime with timezone.  It should return a naive datetime (tzinfo == None)